ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/AuthHandlers.java

```diff
@@ -113,15 +140,10 @@ private static Optional<AuthHandler> getJaasAuthHandler(final Server server) {
     }
   }
 
-  private static AuthHandler basicAuthHandler(final Server server) {
-    final AuthProvider authProvider = new JaasAuthProvider(server, server.getConfig());
+  private static BasicAuthHandler basicAuthHandler(final Server server) {
     final String realm = server.getConfig().getString(KsqlRestConfig.AUTHENTICATION_REALM_CONFIG);
-    final AuthHandler basicAuthHandler = BasicAuthHandler.create(authProvider, realm);
-    // It doesn't matter what we set here as we actually do the authorisation at the
-    // authentication stage and cache the result, but we must add an authority or
-    // no authorisation will be done
-    basicAuthHandler.addAuthority("ksql");
```

Comment:
Note for posterity: the reason it's OK to remove this is because OrHandlerImplementation (called indirectly from RoleBasedAuthZHandler) will properly fail if no roles are provided.